File: 718-Maximum Length of Repeated Subarray.py
class Solution(object):
    def findLength(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: int
        """
        n1, n2 = len(nums1), len(nums2)
        
        dp = [[0]*(n1+1) for _ in range(n2+1)]
        res = 0
        
        for r in range(1, n2+1):
            for c in range(1, n1+1):
                if nums1[c-1] == nums2[r-1]:
                    dp[r][c] = dp[r-1][c-1] + 1
                    res = max(dp[r][c], res)
                    
        return res

File: JavaScript/718-Maximum Length of Repeated Subarray.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var findLength = function(nums1, nums2) {
    const n1 = nums1.length, n2 = nums2.length;
    let res = 0;
    let dp = new Array(n2+1).fill().map(()=>new Array(n1+1).fill(0));
    
    for (let r=1; r<=n2; r++){
        for (let c=1; c<=n1; c++){
            if (nums1[c-1] === nums2[r-1]){
                dp[r][c] = dp[r-1][c-1]+1;
                res = Math.max(dp[r][c], res);
            }
        }
    }
    return res;
};